contrib/lkt/language/parser.py

```diff
@@ -647,6 +647,26 @@ def designated_scope():
             default_val=Entity.check_expr_type.type_scope
         )
 
+    @langkit_property(return_type=T.TypeDecl.entity)
+    def array_element_type():
+        # ArrayLiteral elements type is its parent expected type's actuals
+        # type. As class Array takes only one type, actuals.at(0) is the
+        # ArrayLiteral elements type.
+        return Let(
+            lambda etype=Entity.expected_type.cast(T.InstantiatedGenericType):
+            Cond(
+                # No expected type found
+                etype.is_null,
+                No(T.TypeDecl.entity),
+
+                etype.get_inner_type.as_entity == Entity.array_gen_type,
+                etype.get_actuals.at_or_raise(0),
+
+                # Unreachable
```

Comment:
Shouldn’t it be reachable, for instance when calling this property on an expression that returns a ``LexicalEnv[RootNode]``? I suppose this can happen when trying to analyze `val a: LexicalEnv[RootType] = [1]`.

Reply:
you're right!